### ISSUE TITLE
NVDA can't access context menus opened on some links/images in Edge, even when Narrator can

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#15309]
### Steps to reproduce:

1. Open Microsoft Edge
2. Browse to, for example, [ebary.com](https://ebary.com/)
3. Press `t` to go to the first table, then `k` to go to the second link (which is probably also a graphic).
4. Press `Applications` or `shift+F10`

### Actual behavior:

NVDA speaks the page title and the link, but does not speak the first context menu item.

Note that this does not always happen--sometimes, with no apparent reason, NVDA will send focus to the context menu. Other times it won't.

### Expected behavior:

NVDA should always focus the context menu.
Narrator does.
In fact, if you try this and NVDA doesn't speak the menu, turning on Narrator simultaneously will rectify the problem, and NVDA is able to see the menu.

### Further context

This has been reported on mailing lists for a few months now. I'm not sure why I can't find an issue for it here, as I thought we had one, but I can only find the Edge context menu not opening for the downloads area issue, which @leonardder has a PR to fix now.

### NVDA logs, crash dumps and other attachments:

In the attached log, the first time I try to open a context menu, it fails. From the log, it appears that it starts to read the menu, but is then redirected to the virtual buffer.

Then I try with the next link NVDA takes me to, which also fails, but differently. The second one fails, I believe, because while NVDA is reading the virtual buffer and claiming the page is what I'm navigating, in fact the context menu is open, and that's actually what I'm navigating.

The third link does the same thing, but then I press `escape`, and try to open the context menu on the third link again. This time it succeeds.

[bad_context_edge.txt](https://github.com/nvaccess/nvda/files/12376359/bad_context_edge.txt)

#### NVDA installed/portable/running from source:

Installed, but have tried portables and PR builds as temporaries, with the same effect.

#### NVDA version:

alpha-28879,a7981cad

#### Windows version:

Microsoft® Windows® Operating System version 10.0.22621.1
(This is 11 Home 22H2, but the same thing happened on 10 Home 22H2)

#### Name and version of other software in use when reproducing the issue:

Microsoft Edge version 115.0.1901.203
(But happened as far back as 113, AFAIK)

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Alphas and productions as far back as 2023.1, maybe even pre-2023.1 versions. This is something Edge introduced, I don't believe it's an NVDA change.

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes

**Comment by XLTechie** [comment]
@leonardder could this potentially be fixed by a tweak to your PR #15300?

**Comment by leonardder** [comment]
I assume you tested this with #15300?
Can you also reproduce this when first disabling browse mode before opening the context menu and/or when using UIA for Edge (advanced setting)? In this case, I think it is more of a IA2 vs UIA clash.

**Comment by XLTechie** [comment]
> I assume you tested this with #15300?

Yes, tried that first, before reverting to alpha to do the report.

Both were with UIA for Edge set to default (when necessary).

> Can you also reproduce this when first disabling browse mode before opening the context menu and/or when using UIA for Edge (advanced setting)?

No: in both cases the menu reads fine. Though normal users are unlikely to do either.

Note: there was an error in my original STR: I thought I was on ebay.com, so wrote that in the STR. Because of Edge's autocomplete, it had actually taken me to ebary.com, and that's where the original STR were performed. I have updated accordingly.

**Comment by leonardder** [comment]
I'm pretty sure that this issue has been there right from the start of NVDA's support for Edge Chromium, though it might have been less prevalent.

**Comment by leonardder** [comment]
I just tracked down the cause of the bug. You can temporarily work around it by enable `automatically set system focus to focusable elements` in browse mode settings.
